internal/mage/engine.go

```diff
@@ -198,18 +163,87 @@ func (t Engine) test(ctx context.Context, race bool) error {
 
 	c = c.Pipeline("engine").Pipeline("test")
 
+	opts := util.DevEngineOpts{
+		ConfigEntries: map[string]string{
+			`registry."registry:5000"`:        "http = true",
+			`registry."privateregistry:5000"`: "http = true",
+		},
+	}
+	devEngine := util.DevEngineContainer(c.Pipeline("dev-engine"), []string{runtime.GOARCH}, util.DefaultDevEngineOpts, opts)[0]
+
+	reg := registry(c)
+
+	_, err = devEngine.Export(ctx, "/tmp/engine.tar")
```

Comment:
Added the comment, and because I was pulling out `core/integration/internal` but still needed a dagger cli binary, I decided to use your idea to put it on the filesystem, rather that in a registry.